service/src/main/java/dk/kvalitetsit/stakit/dao/AnnouncementDaoImpl.java

```diff
@@ -108,4 +109,25 @@ public Optional<AnnouncementEntity> getById(long announcementId) {
             return Optional.empty();
         }
     }
+
+    @Override
+    public List<AnnouncementEntity> getAnnouncementsToSend() {
+        var sql = "select * from announcement where is_sent = 0 and :from_datetime >= from_datetime";
```

Comment:
Jeg synes vi skal udvide sql med "for update". I de fleste tilfælde gør det at der ikke er to instanser af servicen der kan selecte de samme rækker på samme tid og dermed få sendt de samme mails to gange.